memory-viz/src/memory_model.ts

```diff
@@ -130,30 +146,44 @@ export class MemoryModel {
      * Clear a given canvas object.
      * @param canvas - the element that is currently used to draw graphics
      */
-    clear(canvas) {
+    clear(canvas: HTMLCanvasElement): void {
         const ctx = canvas.getContext("2d");
         ctx.clearRect(0, 0, canvas.width, canvas.height);
     }
 
     /**
      * Distribute the object drawing depending on type
-     * @param {number} x - value for x coordinate of top left corner
-     * @param {number} y - value for y coordinate of top left corner
-     * @param {string} type - the data type (e.g. list, int) of the object we want draw
-     * @param {number} id - the hypothetical memory address number
-     * @param {*} value - can be passed as a list if type is a collection type
-     * @param {boolean} show_indexes - whether to show list indices
-     * @param {Object} style - The style configuration for the drawings on the canvas (e.g. highlighting, bold texts)
+     * @param x - value for x coordinate of top left corner
+     * @param y - value for y coordinate of top left corner
+     * @param type - the data type (e.g. list, int) of the object we want draw
+     * @param id - the hypothetical memory address number
+     * @param value - can be passed as a list if type is a collection type
+     * @param show_indexes - whether to show list indices
+     * @param style - The style configuration for the drawings on the canvas (e.g. highlighting, bold texts)
      * For style, firstly refer to `style.md` and `presets.md`. For the styling options in terms of texts, refer to
      * the SVG documentation. For the styling options in terms of boxes, refer to the Rough.js documentation.
      */
-    drawObject(x, y, type, id, value, show_indexes, style) {
+    drawObject(
+        x: number,
+        y: number,
+        type: string,
+        id: number,
+        value: object | number[] | string | boolean | null,
+        show_indexes: boolean,
+        style: Style
+    ): Rect {
         if (collections.includes(type)) {
-            if (type === "dict") {
+            if (type === "dict" && typeof value === "object") {
```

Comment:
This check and the one below is good, but it seems like this function might now reach the end of its body without an explicit return. You can raise some kind of error in this case.